### PR TITLE
Discard trace warmup outputs

### PR DESCRIPTION
**Pull request by vkovacevicTT**
### Problem description
Llama3.1-70B was [failing](https://github.com/tenstorrent/tt-xla/actions/runs/23825530884/job/69447587067) in tt-xla LLMBOX benchmarks due to DRAM OOM.

One of the main causes is that during trace capture:

```
func.func private @run_and_capture_trace_3_main(...)
    %0 = "ttnn.get_device"()
    %1 = "ttnn.empty"(%0)
    %2 = "ttnn.empty"(%0)
    "ttnn.write_tensor"(%arg0, %1)
    "ttnn.write_tensor"(%arg1, %2)
    %3:3 = call @trace_3_main(...)                            //warmup
    %4 = "ttnn.begin_trace_capture"(%0)
    %5:3 = call @trace_3_main(...)                            //capture
    "ttnn.end_trace_capture"(%0, %4)
    return %4, %3#0, %3#1, %3#2, [input_slots...], %5#0, %5#1, %5#2
```

both warmup outputs (%3:3 - real, computed values of first invocation) and capture output slots (%5:3 - output slots for future execute trace results, arbitrary values) were used, serving different purposes.

Llama3.1-70B outputs are: 
- cache_position `tensor<1xsi32>`
- logits per user `tensor<32x18x128256xbf16>` - huge for prefill (sequence_len=18) !
- next_token_ids per user `tensor<32x1xsi32>`

Proposed solution is to discard warmup outputs and use trace output slots as outputs for first model run.
In order to fill output slots with actual computed values, trace is executed immediately after capture.

### What's changed
Discarded warmup outputs as return values for `run_and_capture_trace` function.
Added `execute_trace` after capture to populate trace output slots with real values.

Example of IR after change:

```
func.func private @run_and_capture_trace_1_main
    %0 = "ttnn.get_device"()
    %1 = "ttnn.empty"(%0)
    %2 = "ttnn.empty"(%0)
    "ttnn.write_tensor"(%arg0, %1)
    "ttnn.write_tensor"(%arg1, %2)
    %3:3 = call @trace_3_main(...)
    "ttnn.deallocate"(%3#1)
    "ttnn.deallocate"(%3#0)
    "ttnn.deallocate"(%3#2)
    %4 = "ttnn.begin_trace_capture"(%0)
    %5:3 = call @trace_3_main(...)
    "ttnn.end_trace_capture"(%0, %4)
    "ttnn.execute_trace"(%0, %4)
    return %4, [input_slots], [output_slots with real values from first invocation]
```

### Checklist
- [ ] tt-xla benchmarks
    - [x] [Llama3.1-70B on LLMBOX](https://github.com/tenstorrent/tt-xla/actions/runs/24465776763/job/71494762947)
- [ ] tt-xla model tests

### CODE REVIEW

File: test/ttmlir/Dialect/TTNN/trace/scenarios/single_add_no_consteval.mlir [hunk 1]
```diff
@@ -9,8 +9,10 @@ module {
   // CHECK: "ttnn.write_tensor"
```
Comment (sdjordjevicTT) [inline test/ttmlir/Dialect/TTNN/trace/scenarios/single_add_no_consteval.mlir:9]: Can you add a deallocate check in this lit test as well:
test/ttmlir/Dialect/TTNN/trace/scenarios/multichip_kv_cache.mlir

**Review by sdjordjevicTT — APPROVED** [review]
LGTM!

File: lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp [hunk 2]
```diff
@@ -4620,6 +4607,13 @@ class CaptureOrExecuteTraceOpConversionPattern
             getResult.getResult(0));
       }
 
+      // Return the output slots as the actual outputs for the first call.
+      for (size_t i = 0; i < returnVariable.size(); ++i) {
```
Comment (amilovanovicTT) [inline lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp]: It would be nice to merge these two loops into one and omit the need for global variable loading.

File: test/ttmlir/Dialect/TTNN/trace/scenarios/single_add_no_consteval.mlir (hunk 1 above)
Comment (amilovanovicTT) [reply test/ttmlir/Dialect/TTNN/trace/scenarios/single_add_no_consteval.mlir:9]: Also, can you update test/ttmlir/Dialect/TTNN/trace/negative_capture_or_execute_trace.mlir?

**Review by amilovanovicTT — APPROVED** [review]

File: lib/Dialect/TTNN/Transforms/TTNNTraceHoistTransform.cpp [hunk 3]
```diff
@@ -390,28 +390,23 @@ class TTNNTraceHoistTransform
 
     // The capture function returns multiple values for trace management:
     // 1. traceId - Identifier for the captured trace
-    // 2. actual outputs - Results from the first execution (non-traced)
-    // 3. trace input slots - Persistent device memory for input data
-    // 4. trace output slots - Persistent device memory for output data
+    // 2. trace input slots - Persistent device memory for input data
+    // 3. trace output slots - Persistent device memory for output data
+
```
[the remark this answers is not in the document]
Comment (vkovacevicTT) [reply lib/Dialect/TTNN/Transforms/TTNNTraceHoistTransform.cpp:395]: Nice catch bro @copilot

**Review by pilkicTT — APPROVED** [review]
